Sources/StreamChat/Database/DatabaseContainer_Mock.swift

```diff
@@ -71,6 +74,19 @@ class DatabaseContainerMock: DatabaseContainer {
         resetEphemeralValues_called = true
         super.resetEphemeralValues()
     }
+    
+    override func loadPersistentStores(completionHandler block: @escaping (NSPersistentStoreDescription, Error?) -> Void) {
+        // If shouldFailLoadingStore is true and the recreating of the store has not been called yet, it should return error
+        if shouldFailLoadingStore, !recreatePersistentStore_called {
+            super.loadPersistentStores { store, _ in
+                block(store, NSError(domain: "test", code: 1, userInfo: nil))
```

Comment:
nit: You can use `TestError()` for cases like this.

Reply:
> A huge 👍 for creativity when figuring out how to test this. The only thing I'm afraid of is that the test uses too much knowledge of the inner working of `DatabaseContainer`.
> 
> What if we structured the test a bit differently:
> 
> 1. Create two different CoreData models (like we do with `TestDataModel` in other tests) and have them completely different (i.e. containing a different entity)
> 2. We create `DatabaseContainer` with the first model
> 3. We try to create another container with the second model but using the same URL
> 4. We just need to check the container is created correctly.

Yes, I agree, when the system under test is also being mocked, it is very difficult to test some stuff. In this case, your suggestion is better since it reflects the real use case better than the creative approach 👍 Also, adding logic to mocks is a red flag (but I case in this case there wouldn't be another way)

@VojtaStavik Please re-review :)